contrib/platform/src/com/sun/jna/platform/win32/Advapi32.java

```diff
@@ -1175,6 +1175,61 @@ public boolean ReadEventLog(HANDLE hEventLog, int dwReadFlags,
 	public boolean GetOldestEventLogRecord(HANDLE hEventLog,
 			IntByReference OldestRecord);
 
+
+	/**
+	 * Changes the optional configuration parameters of a service.
+	 *
+	 * @param hService
+	 *            A handle to the service. This handle is returned by the
+	 *            OpenService or CreateService function and must have the
+	 *            SERVICE_CHANGE_CONFIG access right. For more information,
+	 *            see <a
+	 *            href="http://msdn.microsoft.com/en-us/library/ms685981.aspx"
+	 *            >Service Security and Access Rights</a>.
+	 *            If the service controller handles the SC_ACTION_RESTART
+	 *            action, hService must have the SERVICE_START access right.
+	 * @param dwInfoLevel
+	 *            The configuration information to be changed.
+	 * @param lpInfo
+	 *            A pointer to the new value to be set for the configuration
+	 *            information. The format of this data depends on the value
+	 *            of the dwInfoLevel parameter. If this value is NULL, the
+	 *            information remains unchanged.
+	 * @return If the function succeeds, the return value is nonzero.
+	 *         If the function fails, the return value is zero. To get extended
+	 *         error information, call GetLastError.
+	 */
+	public boolean ChangeServiceConfig2(SC_HANDLE hService, int dwInfoLevel,
+			Structure lpInfo);
```

Comment:
Using the generic _Structure_ is a bit too general - whenever the API expects several variants we "overload" the method - see _GetVersionEx_ as an example. In this case there should be a match between the _dwInfoLevel_ and the actual structure type, but it is up to the user to make sure that the level matches the type. Another alternative would be to define an **abstract** _ServiceConfigStructure_ that extends _Structure_ and have all the possible variants extend the _ServiceConfigStructure_ - this would allow some degree of type safety...